src/mono/mono/mini/interp/jiterpreter.c

```diff
@@ -925,6 +928,7 @@ mono_interp_tier_prepare_jiterpreter_fast (
 #ifdef DISABLE_THREADS
 	gint64 count = trace_info->hit_count++;
 #else
+	static_assert (sizeof(atomic_long) == sizeof(gint64) && ATOMIC_LONG_LOCK_FREE == 2);
 	gint64 count = atomic_fetch_add ((atomic_long *)&trace_info->hit_count, 1);
```

Comment:
@kg PTAL at this file.

This particular use of `atomic _long` is actually kind of suspect.  on some platforms it's actually `long long` (and `atomic_llong`) that is 64-bits.  I assume that's not the case on wasm?

Reply:
Yeah this might need to be atomic_llong. I did test it originally but it's possible it's truncating it.

Reply:
```
/__w/1/s/src/mono/mono/mini/interp/jiterpreter.c:931:2: error: static assertion failed due to requirement 'sizeof(_Atomic(long)) == sizeof (trace_info->hit_count)': 
          static_assert (sizeof(atomic_long) == sizeof(trace_info->hit_count) && ATOMIC_LONG_LOCK_FREE == 2, "");
          ^              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /usr/local/emscripten/emsdk/upstream/emscripten/cache/sysroot/include/assert.h:12:23: note: expanded from macro 'static_assert'
  #define static_assert _Static_assert
                        ^
  /__w/1/s/src/mono/mono/mini/interp/jiterpreter.c:931:37: note: expression evaluates to '4 == 8'
```

yep. looks like it needs to be `atomic_llong`